catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/CachedSource.java

```diff
@@ -197,4 +202,30 @@ public String getAttributeValue(String attr) {
     private void clearContentTypes() {
         this.cachedContentTypes = Collections.<ContentType>emptySet();
     }
+
+    public boolean equals(Object cachedSource) {
+        if (cachedSource == null) {
+            return false;
+        }
+        if (!(cachedSource instanceof CachedSource)) {
+            return false;
+        }
+        CachedSource tmpSource = (CachedSource) cachedSource;
+        Comparator<CachedSource> sourceKeyComparator = Comparator.comparing(
+                CachedSource::getId,
+                Comparator.nullsLast(Comparator.naturalOrder()))
+                .thenComparing(CachedSource::getTitle,
+                        Comparator.nullsLast(Comparator.naturalOrder()))
+                .thenComparing(CachedSource::getVersion,
+                        Comparator.nullsLast(Comparator.naturalOrder()))
+                .thenComparing(CachedSource::getDescription,
+                        Comparator.nullsLast(Comparator.naturalOrder()))
+                .thenComparing(CachedSource::getOrganization,
+                        Comparator.nullsLast(Comparator.naturalOrder()));
+        return sourceKeyComparator.compare(this, tmpSource) == 0;
+    }
+
+    public int hashCode() {
+        return uniqueId.hashCode();
```

Comment:
I believe this and your equals() method could give you different results. If you're going to use the UUID for its hashcode, your equals method should test that for equality.